easybuild/easyblocks/generic/systemmpi.py

```diff
@@ -0,0 +1,255 @@
+            raise EasyBuildError("Path derived for system MPI (%s) does not exist: %s!", mpi_name, self.mpi_prefix)
+
+        self.log.debug("Derived version/install prefix for system MPI %s: %s, %s",
+                       mpi_name, self.mpi_version, self.mpi_prefix)
+
+        # For the version of the underlying C compiler need to explicitly extract (to be certain)
+        self.c_compiler_version = extract_compiler_version(self.mpi_c_compiler)
+        self.log.debug("Derived compiler/version for C compiler underneath system MPI %s: %s, %s",
+                       mpi_name, self.mpi_c_compiler, self.c_compiler_version)
+
+        # If EasyConfig specified "real" version (not 'system' which means 'derive automatically'), check it
+        if self.cfg['version'] == 'system':
+            self.log.info("Found specified version '%s', going with derived MPI version '%s'",
+                          self.cfg['version'], self.mpi_version)
+        elif self.cfg['version'] == self.mpi_version:
+            self.log.info("Specified MPI version %s matches found version" % self.mpi_version)
+        else:
+            raise EasyBuildError("Specified version (%s) does not match version reported by MPI (%s)" %
+                                 (self.cfg['version'], self.mpi_version))
+
+        # fix installdir and module names (may differ because of changes to version)
+        mns = ActiveMNS()
+        self.cfg.full_mod_name = mns.det_full_module_name(self.cfg)
+        self.cfg.short_mod_name = mns.det_short_module_name(self.cfg)
+        self.cfg.mod_subdir = mns.det_module_subdir(self.cfg)
+
+        # keep track of original values, for restoring later
+        self.orig_version = self.cfg['version']
+        self.orig_installdir = self.installdir
+
+    def make_installdir(self, dontcreate=None):
+        """Custom implementation of make installdir: do nothing, do not touch system MPI directories and files."""
+        pass
+
+    def make_module_req_guess(self):
+        """
+        A dictionary of possible directories to look for.  Return appropriate dict for system MPI.
+        """
+        if self.cfg['name'] ==  "impi":
+            # Need some extra directories for Intel MPI, assuming 64bit here
+            lib_dirs = ['lib/em64t', 'lib64']
+            include_dirs = ['include64']
+            return_dict = {
+                'PATH': ['bin/intel64', 'bin64'],
+                'LD_LIBRARY_PATH': lib_dirs,
+                'LIBRARY_PATH': lib_dirs,
+                'CPATH': include_dirs,
+                'MIC_LD_LIBRARY_PATH': ['mic/lib'],
+            }
+        else:
+            return_dict = {}
+
+        return return_dict
+
+    def make_module_step(self, fake=False):
+        """
+        Custom module step for SystemMPI: make 'EBROOT' and 'EBVERSION' reflect actual system MPI version
+        and install path.
+        """
+        # First let's verify that the toolchain and the compilers under MPI match
+        if self.travis:
```

Comment:
Are we reimplementing https://github.com/auchenberg/volkswagen ?

This kind of code should never make it into easyblocks; instead, we should modify the tests to test specific easyblocks under controlled circumstances. See how we handle `systemcompiler.py` in `test/easyblocks/init_easyblocks.py`